homeassistant/components/osoenergy/sensor.py

```diff
@@ -0,0 +1,181 @@
+    for hour in range(24):
+        local_hour = _get_local_hour(hour)
+        profile[local_hour.hour] = values[hour]
+
+    return profile
+
+
+@dataclass(frozen=True)
+class OSOEnergySensorEntityDescription(SensorEntityDescription):
+    """Class describing OSO Energy heater sensor entities."""
+
+    value: Callable[[OSOEnergy], StateType] = round
+
+
+SENSOR_TYPES: tuple[OSOEnergySensorEntityDescription, ...] = (
+    OSOEnergySensorEntityDescription(
+        key="profile",
+        translation_key="profile",
+        value=lambda device: _convert_profile_to_local(device.state),
+    ),
+    OSOEnergySensorEntityDescription(
+        key="heater_mode",
+        translation_key="heater_mode",
+        device_class=SensorDeviceClass.ENUM,
+        value=lambda device: ENUM_VALUE_MAPPING["heater_mode"].get(
+            device.state.lower(), device.state.lower()
+        ),
+    ),
+    OSOEnergySensorEntityDescription(
+        key="optimization_mode",
+        translation_key="optimization_mode",
+        device_class=SensorDeviceClass.ENUM,
+        value=lambda device: ENUM_VALUE_MAPPING["optimization_mode"].get(
+            device.state.lower(), device.state.lower()
+        ),
+    ),
+    OSOEnergySensorEntityDescription(
+        key="power_load",
+        translation_key="power_load",
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        value=lambda device: device.state,
+    ),
+    OSOEnergySensorEntityDescription(
+        key="tapping_capacity_kwh",
+        translation_key="tapping_capacity_kwh",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        value=lambda device: device.state,
+    ),
+    OSOEnergySensorEntityDescription(
+        key="capacity_mixed_water_40",
+        translation_key="capacity_mixed_water_40",
+        device_class=SensorDeviceClass.VOLUME,
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        value=lambda device: device.state,
+    ),
+    OSOEnergySensorEntityDescription(
+        key="v40_min",
+        translation_key="v40_min",
+        device_class=SensorDeviceClass.VOLUME,
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        value=lambda device: device.state,
+    ),
+    OSOEnergySensorEntityDescription(
+        key="v40_level_min",
+        translation_key="v40_level_min",
+        device_class=SensorDeviceClass.VOLUME,
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        value=lambda device: device.state,
+    ),
+    OSOEnergySensorEntityDescription(
+        key="v40_level_max",
+        translation_key="v40_level_max",
+        device_class=SensorDeviceClass.VOLUME,
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        value=lambda device: device.state,
+    ),
+    OSOEnergySensorEntityDescription(
+        key="volume",
+        translation_key="volume",
```

Comment:
Can use device class translations. So if there is no name or translation key set it will use the name of the device class
```suggestion
```